calculator.rb

```diff
@@ -0,0 +1,71 @@
+def add(num1, num2)
+  num1 + num2
+end
+
+def subtract(num1, num2)
+  num1 - num2
+end
+
+def divide(num1, num2)
+  if num2 != 0
+    num1 / num2
+  else
+    puts "What are you doing? You can't divide by zero!"
+    return nil
+  end
+end
+
+def multiply(num1, num2)
+  num1 * num2
+end
+
+def verify(input)
+  input = gets.chomp
+  case input 
+  when "0"
+    input.to_i
+  when /\d/
+    input.to_i
+  else
+    puts "Please enter a number: "
+    verify(input)
+  end
```

Comment:
Here, `verify` calls itself! A method calling itself is a programming technique called recursion, which you'll learn about in CS Fun in a few months.

Recursion is a powerful tool that can elegantly solve many problems. However, in this case I think a `while` loop might be a better choice, if only because it makes it immediately obvious to the reader that this code might execute many times. Re-writing this code with a `while` loop might look like:

```ruby
def verify(input)
  while true
    input = gets.chomp
    case input 
    when "0"
      return input.to_i
    when /\d/
      return input.to_i
    else
      puts "Please enter a number: "
    end
  end
end
```

What you've written certainly solves the problem - I just want to put a name on it.